src/pages/dex/[poolpairId]/index.page.tsx

```diff
@@ -67,15 +68,47 @@ export default function PoolPairPage (props: InferGetServerSidePropsType<typeof
 }
 
 export async function getServerSideProps (context: GetServerSidePropsContext): Promise<GetServerSidePropsResult<PoolPairPageProps>> {
-  const poolpairId = context.params?.poolpairId?.toString().trim() as string
+  let poolpairId = context.params?.poolpairId?.toString().trim() as string
+  if (!isAlphanumeric(poolpairId, '-')) {
+    return { notFound: true }
+  }
+
   const api = getWhaleApiClient(context)
 
+  let poolPair: PoolPairData
+
+  if (poolpairId.includes('-')) {
+    const [tokenA, tokenB] = poolpairId.split('-')
+    const poolpairs: PoolPairData[] = []
+
+    let poolpairsResponse = await api.poolpairs.list(200)
+    poolpairs.push(...poolpairsResponse)
+    while (poolpairsResponse.hasNext) {
+      poolpairsResponse = await api.poolpairs.list(200, poolpairsResponse.nextToken)
+      poolpairs.push(...poolpairsResponse)
+    }
+    poolPair = poolpairs.filter(pair => {
+      return pair.tokenA.displaySymbol === tokenA && pair.tokenB.displaySymbol === tokenB
+    }).pop()!
```

Comment:
Try not to use `!`. Add undefined to the variable type instead and check for it

Reply:
We should shift this whole if chunk into a function. It is going to get more complex when we add in the other url types.

Reply:
Pushed code to split logic into a separate async function